Add EN Doc: Uninstall Pluggable Components

Felixnoo opened the pull request:
Signed-off-by: Felixnoo <felixliu@yunify.com>

Fix #1591

Felixnoo commented:
/assign @zheng1 @LinuxSuRen @benjaminhuo @wanjunlei @zryfish 
Hi, I'm adding a doc about the uninstallation of each pluggable component for KS v3.1.1. **Service Topology** and **Pod IP Pools** are missing uninstallation methods.
Please help review this doc. Thanks.

Felixnoo commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,176 @@
+
+{{< notice note >}}
+
+This operation will make Auditing, Events, and Service Mesh unavailable.
```
@wanjunlei 
Hi, I think `This operation will affect auditing, events, and Service Mesh.` is a little bit vague. So I made a small change to your original sentence. Does it look good to you?

yuswift commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
````diff
@@ -0,0 +1,176 @@
+## Uninstall KubeSphere App Store
+
+```bash
+kubectl delete ns openpitrix-system
````
In v3.1, we don't need to delete the ns anymore.

Felixnoo replied:
So we just change `true` to `false` for `openpitrix.store.enabled` in the cc, then the App Store will be uninstalled?

yuswift replied:
Yeap. BTW, the docs for v3.1 and v3.0 may be different. I guess we need to point that out.

yuswift commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
````diff
@@ -0,0 +1,176 @@
+
+## Uninstall KubeSphere DevOps
+
+```bash
````
I think we should disable the component first in the cc. So does the other component.

Felixnoo replied:
OK.

Felixnoo replied:
[The uninstallation guide for KS v3.0.0](https://v3-0.docs.kubesphere.io/docs/faq/installation/uninstall-pluggable-components/) is missing this part of editing cc. I guess this is also applicable to it?

Felixnoo commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,215 @@
+
+   {{< notice note >}}
+
+   This operation will make Auditing, Events, and Service Mesh unavailable.
```
@wanjunlei 
Hi, I think `This operation will affect auditing, events, and Service Mesh.` is a little bit vague. So I made some changes to your original sentence. Does it look good to you?

Felixnoo replied:
Wording updated in [this commit](https://github.com/kubesphere/website/pull/1863/commits/d08505237dff4fcbdd8306e1cc274ab3ff8a6745).

zryfish commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,215 @@
+
+## Uninstall KubeSphere App Store
+
+Change the value of `openpitrix.store.enabled` from `true` to `false` in `ks-installer`.
```
`ks-installer` is unclear, better to use `ClusterConfiguration ks-installer`. There are two methods to change it:
1. `kubectl -n kubesphere-system edit clusterconfiguration ks-installer`
2. from UI
 ![截屏2021-08-16 11 28 09](https://user-images.githubusercontent.com/3326354/129507755-cc6c77e5-e413-408b-a3d4-f3c23f4baf56.png)

Felixnoo replied:
Thanks for your comment. Updates made in [this commit](https://github.com/kubesphere/website/pull/1863/commits/5fdeba6088caf26e9be3fdef124656cf6f636a22).

zryfish commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,215 @@
+
+## Uninstall KubeSphere DevOps
+
+1. Change the value of `devops.enabled` from `true` to `false` in `ks-installer`.
```
It's better to wait for the `ks-installer` pod to complete to proceed.

zryfish commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,215 @@
+
+   {{< notice note >}}
+
+   After running this command, KubeSphere stops collecting container logs.
```
You still can view container recent logs, only you are not able to browse history logs.

Felixnoo replied:
Updates made [here](https://github.com/kubesphere/website/pull/1863/commits/5fdeba6088caf26e9be3fdef124656cf6f636a22#diff-8c43bfc4100d51031fa4c69b1e26b3005862122b8f91b7410de3408072d66b0dR103).

junotx commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
````diff
@@ -0,0 +1,215 @@
+   helm delete ks-events -n kubesphere-logging-system
+   ```
+
+## Uninstall KubeSphere Alerting and Notification
````
Perfom the following steps to uninstall Alerting for KubeSphere v3.1.x: 
1. Change the value of `alerting.enabled` from `true` to `false` in `ks-installer`.
2. Run the following command:
    ```bash
    kubectl -n kubesphere-monitoring-system delete thanosruler kubesphere
    ```
In addition, Notification is installed by default and may no longer be enabled or disabled

Felixnoo replied:
Thanks for your comment. Updates made in [this commit](https://github.com/kubesphere/website/pull/1863/commits/d48ea46b0fb97537a354d11ef92671abb5f35f5f).

RolandMa1986 commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,218 @@
+   curl -L https://istio.io/downloadIstio | sh -
+   istioctl x uninstall --purge
+   
+   kubectl -n istio-system delete kiali
```
The delete command should be:
`kubectl -n istio-system delete kiali kiali`

Felixnoo replied:
Thanks for your comment. Updates made in [this commit](https://github.com/kubesphere/website/pull/1863/commits/b1d0432ebb9a06e07d329f6aa4c3b3bc04a70f8c).

RolandMa1986 commented:
/lgtm

JohnNiang commented:
/ping @kubesphere/sig-devops

junotx commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
````diff
@@ -0,0 +1,219 @@
+
+For the component NetworkPolicy, disabling it does not require uninstalling the component as its controller is now inside `ks-controller-manager`. If you want to remove it from the KubeSphere console, change the value of `network.networkpolicy.enabled` from `true` to `false` in `ks-installer` of the CRD `ClusterConfiguration`.
+
+## Uninstall Metrics Server
+
+1. Change the value of `metrics_server.enabled` from `true` to `false` in `ks-installer` of the CRD `ClusterConfiguration`.
+
+2. Run the following command:
+
+   ```bash
+   helm delete metrics-server -n kube-system
+   ```
````
Perfom the following steps to uninstall Metrics Server for KubeSphere v3.1.x: 
1. Change the value of `metrics_server.enabled` from `true` to `false` in `ks-installer` of the CRD `ClusterConfiguration`.
2. Run the following commands: 
    ```shell
    kubectl delete apiservice v1beta1.metrics.k8s.io
    kubectl -n kube-system delete service metrics-server
    kubectl -n kube-system delete deployment metrics-server
    ```

Felixnoo replied:
Thanks! Updates made in [this commit](https://github.com/kubesphere/website/pull/1863/commits/a073b38fc68dcd83cc9ca330c57de8933922aee7).

junotx commented:
The Events uninstallation is ok for me

Felixnoo commented:
I updated the [uninstallation methods](https://github.com/kubesphere/website/pull/1863/commits/6c1b6c81009d4763b0085f44cc0612c6add2ca24) for **Service Topology** and **Pod IP Pools** based on my own understanding. Please review them.

zheng1 commented:
> I updated the [uninstallation methods](https://github.com/kubesphere/website/pull/1863/commits/6c1b6c81009d4763b0085f44cc0612c6add2ca24) for **Service Topology** and **Pod IP Pools** based on my own understanding. Please review them.

@Felixnoo  **Pod IP Pools** only the configuration needs to be modified, no data needs to be deleted

Felixnoo commented:
> > I updated the [uninstallation methods](https://github.com/kubesphere/website/pull/1863/commits/6c1b6c81009d4763b0085f44cc0612c6add2ca24) for **Service Topology** and **Pod IP Pools** based on my own understanding. Please review them.
> 
> @Felixnoo **Pod IP Pools** only the configuration needs to be modified, no data needs to be deleted

Thanks! Updated in [this commit](https://github.com/kubesphere/website/pull/1863/commits/ff9c9fd76df80becc35cf3361c3640861bba6846).

Felixnoo commented:
Still needs comments on **DevOps** and **Service Topology**.

JohnNiang commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,227 @@
+
+## Uninstall KubeSphere DevOps
+
+1. Change the value of `devops.enabled` from `true` to `false` in `ks-installer` of the CRD `ClusterConfiguration`.
```
How about deleting `status.devops` at the same time?

Felixnoo replied:
Thanks for your comment. I'm not sure if this is required, but I'm with you if you are sure about this. So, where can I delete `status.devops`?

Felixnoo replied:
Hum... I just found out that only part of the YAML file of `ks-installer` is shown on the console. The `status.devops` part is not shown.
<img width="1234" alt="Xnip2021-08-20_09-49-13" src="https://user-images.githubusercontent.com/75110798/130166454-6bf2cead-3a93-40a7-afe4-48bded12bbbf.png">

But I can see that part on CLI. Additionally, if DevOps needs to delete the `status.devops` part, will that apply to other components? 
<img width="839" alt="Xnip2021-08-20_09-53-38" src="https://user-images.githubusercontent.com/75110798/130166695-3fdf66ab-e68d-4c5f-89c0-7348f9a98953.png">

JohnNiang replied:
If the status is not deleted, users cannot enable DevOps again by setting devops.enable to true.

Felixnoo replied:
Sure. I'll update the doc.

Felixnoo replied:
> Hum... I just found out that only part of the YAML file of `ks-installer` is shown on the console. The `status.devops` part is not shown.
> <img alt="Xnip2021-08-20_09-49-13" width="1234" src="https://user-images.githubusercontent.com/75110798/130166454-6bf2cead-3a93-40a7-afe4-48bded12bbbf.png">
> 
> But I can see that part on CLI. Additionally, if DevOps needs to delete the `status.devops` part, will that apply to other components?
> ![Xnip2021-08-20_09-53-38](https://user-images.githubusercontent.com/75110798/130167662-edec2d59-f56f-4008-b1fb-2b02c13e229f.png)

Hi @zryfish, I'm thinking if the DevOps component requires the deletion of `status.devops`, do other components also need to do delete their corresponding part?

Felixnoo replied:
> Hum... I just found out that only part of the YAML file of `ks-installer` is shown on the console. The `status.devops` part is not shown.
> <img alt="Xnip2021-08-20_09-49-13" width="1234" src="https://user-images.githubusercontent.com/75110798/130166454-6bf2cead-3a93-40a7-afe4-48bded12bbbf.png">
> 
> But I can see that part on CLI. Additionally, if DevOps needs to delete the `status.devops` part, will that apply to other components?
> ![Xnip2021-08-20_09-53-38](https://user-images.githubusercontent.com/75110798/130167662-edec2d59-f56f-4008-b1fb-2b02c13e229f.png)

Since other components did not mention anything about the deletion of `status.XXX`, I'll take it that other components do not require this operation. Please let me know if there's any further updates.

Felixnoo replied:
Since other components did not mention anything about the deletion of `status.XXX`, I'll take it that other components do not require this operation. Please let me know if there are any further updates.

zheng1 commented:
@Felixnoo   LGTM for Service Topology

Felixnoo commented:
> @Felixnoo LGTM for Service Topology

OK, many thanks~

Felixnoo commented:
Since other components did not mention anything about the deletion of status.XXX, I'll take it that other components do not require this operation. Please let me know if there are any further updates.

Felixnoo commented:
/assign @Patrick-LuoYu 
Hi, please take a look at this PR, thanks.

FeynmanZhou commented on content/en/docs/faq/installation/uninstall-pluggable-components.md:
```diff
@@ -0,0 +1,229 @@
+linkTitle: "Uninstall Pluggable Components from KubeSphere v3.1.x"
+Weight: 16500
+---
+
```
Is it better to divide this guide into different sections and add them to [Enable Pluggable Components](https://kubesphere.io/docs/pluggable-components/)? IMO, FAQ is not a suitable place for this guide.

Felixnoo replied:
Thanks for your comment. I chose to put this guide as a whole in FAQ for v3.1.x and our team will refactor the documentation structure in the next release. Is that OK?

FeynmanZhou replied:
@Patrick-LuoYu What's your opinion?

Felixnoo replied:
Moved doc to the chapter Enable Pluggable Components.

Patrick-LuoYu commented:
/lgtm
/approve
